ansq/tcp/connection.py

```diff
@@ -476,6 +500,7 @@ async def open_connection(
     on_exception: Callable = None,
     loop: AbstractEventLoop = None,
     auto_reconnect: bool = True,
+    auto_reconnect_interval: int = 15000,
```

Comment:
I would use `auto_reconnect_interval= 15s`, but made it to be consistent with `heartbeat_interval`.

Reply:
IMO, in many usecases 15 seconds is too long to wait for a reconnect. On the other hand something like 1s is another extreme. In ideal world I'd like to see a progressive interval. Something like a geometric progression. Say, 2s, 4s, 8s, 16s and so on for consequent retries. Maybe up to 32s at max.

Also, the auto reconnect behavior is extensive for a lookupd supported connections. Thus, it applies for a small single instance setups only where one doesn't usually wish to spend a lot of time on tunning.

Reply:
Two different approach is described [here](https://nsq.io/clients/building_client_libraries.html#reconnection) by the way:
- for a single connection there should be progressive intervals
- for a consumer with dynamic connections (lookupd) there should be a constant interval

what would you think?

Reply:
Sounds reasonable. There could be indeed cases when reconnecting is justified regardless of data received from lookups.

Reply:
I've added progression.